storage/src/test/java/org/apache/kafka/server/log/remote/storage/RemoteLogManagerConfigTest.java

```diff
@@ -27,171 +25,72 @@
 
 import static org.apache.kafka.server.log.remote.storage.RemoteLogManagerConfig.DEFAULT_REMOTE_LOG_METADATA_MANAGER_CLASS_NAME;
 import static org.apache.kafka.server.log.remote.storage.RemoteLogManagerConfig.REMOTE_LOG_METADATA_MANAGER_CLASS_NAME_PROP;
+import static org.apache.kafka.server.log.remote.storage.RemoteLogManagerConfig.REMOTE_LOG_METADATA_MANAGER_LISTENER_NAME_PROP;
+import static org.apache.kafka.server.log.remote.storage.RemoteLogManagerConfig.REMOTE_LOG_STORAGE_SYSTEM_ENABLE_PROP;
+import static org.apache.kafka.server.log.remote.storage.RemoteLogManagerConfig.REMOTE_STORAGE_MANAGER_CLASS_PATH_PROP;
 import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertNotEquals;
 
 public class RemoteLogManagerConfigTest {
 
-    private static class TestConfig extends AbstractConfig {
-        public TestConfig(Map<?, ?> originals) {
-            super(RemoteLogManagerConfig.CONFIG_DEF, originals, true);
-        }
-    }
-
     @ParameterizedTest
     @ValueSource(booleans = {true, false})
     public void testValidConfigs(boolean useDefaultRemoteLogMetadataManagerClass) {
         String rsmPrefix = "__custom.rsm.";
         String rlmmPrefix = "__custom.rlmm.";
         Map<String, Object> rsmProps = Collections.singletonMap("rsm.prop", "val");
         Map<String, Object> rlmmProps = Collections.singletonMap("rlmm.prop", "val");
-        RemoteLogManagerConfig expectedRemoteLogManagerConfig = getRemoteLogManagerConfig(useDefaultRemoteLogMetadataManagerClass,
-                rsmPrefix,
-                rlmmPrefix,
-                rsmProps,
-                rlmmProps);
 
-        Map<String, Object> props = extractProps(expectedRemoteLogManagerConfig);
+        Map<String, Object> props = getRLMProps(useDefaultRemoteLogMetadataManagerClass);
         rsmProps.forEach((k, v) -> props.put(rsmPrefix + k, v));
         rlmmProps.forEach((k, v) -> props.put(rlmmPrefix + k, v));
+
+        RemoteLogManagerConfig expectedRemoteLogManagerConfig = new RemoteLogManagerConfig(props);
+
         // Removing remote.log.metadata.manager.class.name so that the default value gets picked up.
         if (useDefaultRemoteLogMetadataManagerClass) {
             props.remove(REMOTE_LOG_METADATA_MANAGER_CLASS_NAME_PROP);
         }
```

Comment:
This is such an incredibly silly test now that we're using AbstractConfig. Of course the default will be applied if the custom value isn't present in the map, that's what AbstractConfig is useful for!

If `useDefaultRemoteLogMetadataManagerClass` is false, then this test reduces to `assertEquals(new RemoteLogManagerConfig(props).values(), new RemoteLogManagerConfig(props).values());` which is also very low-value.

I think removing the parameterization, and confirming that the RLMC can be instantiated once is more than sufficient here.

You should probably add an assertion for the rsmProps and rlmmProps, comparing them to the output of the non-trivial methods in the RLMC, since that is coverage that we are losing in this refactor.

If you want, you can also add a test that constructs an RLMC with an empty configuration, because there are default values specified for all configurations.

If you are really generous with your time, you can also add some bad values and expect ConfigExceptions. A few of the configurations have NonEmptyString validators, so you could verify those are present with tests.

Reply:
Agree, updated and added these tests.